packages/firestore/src/model/document.ts

```diff
@@ -26,7 +26,20 @@ export interface DocumentOptions {
   hasLocalMutations: boolean;
 }
 
-export class Document {
+/**
+ * The result of a lookup for a given path may be an existing document or a
+ * marker that this document does not exist at a given version.
+ */
+export abstract class MaybeDocument {
+  readonly key: DocumentKey;
+  readonly version: SnapshotVersion;
+}
+
+/**
+ * Represents a document in Firestore with a key, version, data and whether the
+ * data has local mutations applied to it.
+ */
+export class Document implements MaybeDocument {
```

Comment:
Shouldn't this be `extends` rather than `implements`?

Here and below.

Reply:
Yeah, I would have hoped that TS complains about things like this.